### PR TITLE
MAYA-125723 - Implement display layer's bbox mode for instanced geometry

### PR DESCRIPTION
**Pull request by vlasovi**

### CODE REVIEW

File: lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.cpp [hunk 1]
```diff
@@ -319,6 +319,16 @@ TfToken MayaUsdRPrim::_GetMaterialNetworkToken(const TfToken& reprToken) const
     return _displayLayerModes._texturing ? reprToken : TfToken();
 }
 
+HdReprSharedPtr MayaUsdRPrim::_FindRepr(const ReprVector& reprs, const TfToken& reprToken)
+{
+    ReprVector::const_iterator it
+        = std::find_if(reprs.begin(), reprs.end(), [reprToken](ReprVector::const_reference e) {
+              return reprToken == e.first;
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.cpp:326]: I suppose the vector is not sorted so std::binary_search cannot be used.

File: lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.cpp [hunk 2]
```diff
@@ -330,38 +340,49 @@ HdReprSharedPtr MayaUsdRPrim::_InitReprCommon(
...
+    auto* const param = static_cast<HdVP2RenderParam*>(_delegate->GetRenderParam());
+    auto&       drawScene = param->GetDrawScene();
+
+    // display layers handling
+    if (!drawScene.GetUsdImagingDelegate()->GetInstancerId(id).IsEmpty()) {
+        _displayLayerModes = DisplayLayerModes();
+
+        // Instanced primitives with instances in display layers use 'forced' representations to
+        // draw those specific instances, so the 'forced' representations should be inited alongside
+        if (reprToken != HdVP2ReprTokens->forcedBbox) {
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.cpp:352]: inited -> initialized, but not worth making a new commit just for that.

File: lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.cpp [hunk 3]
```diff
@@ -330,38 +340,49 @@ HdReprSharedPtr MayaUsdRPrim::_InitReprCommon(
...
 
     _UpdateReprOverrides(reprs);
 
-    // Finally we can get to actually init/dirty the repr
-    ReprVector::const_iterator it
-        = std::find_if(reprs.begin(), reprs.end(), [reprToken](ReprVector::const_reference e) {
-              return reprToken == e.first;
-          });
-    HdReprSharedPtr curRepr = (it != reprs.end() ? it->second : nullptr);
+    // Find the current representation in the array of all inited representations
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.cpp:364]: Same inited -> initialized, again not worth a commit.

File: lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.h [hunk 4]
```diff
@@ -258,7 +260,17 @@ class MayaUsdRPrim
...
+    void _SyncDisplayLayerModesInstanced(SdfPath const& id, unsigned int instanceCount);
+
+    bool _ShouldSkipInstance(unsigned int usdInstanceId, const TfToken& reprToken) const;
+
+    void _SyncForcedReprs(
+        HdRprim&          refThis,
+        HdSceneDelegate*  delegate,
+        HdRenderParam*    renderParam,
+        HdDirtyBits*      dirtyBits,
+        ReprVector const& reprs);
```
Comment (pierrebai-adsk) [inline lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.h:273]: I'd like to have some docs/comment that explain what each function does and why. I guess it falls into a larger scope of documenting how the render delegates work.

**Review by pierrebai-adsk — APPROVED** [review]

File: lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.cpp (hunk 1 above)
Comment (vlasovi) [reply lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.cpp:326]: Those are stored in an RPrim and managed by Hydra. They are not sorted.

**Pull request merged**